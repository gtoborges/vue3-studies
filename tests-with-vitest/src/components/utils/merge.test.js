import { test, expect } from 'vitest'
import { deepMerge } from './merge'

test('shallow merge', () => {
  const merged = deepMerge(
    {
      name: 'Augusto'
    },
    {
      github: 'gtoborges'
    }
  )

  expect(merged).toEqual({
    name: 'Augusto',
    github: 'gtoborges'
  })
})

test('shallow merge with overlaps', () => {
  const merged = deepMerge(
    {
      name: 'Augusto',
      github:'uknown'
    },
    {
      github: 'gtoborges',
      twitter: 'gtoborges_'
    }
  )

  expect(merged).toEqual({
    name: 'Augusto',
    github: 'gtoborges',
    twitter: 'gtoborges_'
  })
})

test('shallow merge with arrays', () => {
  const merged = deepMerge(
    ['vue', 'react'],
    ['svelte', 'solid']
  )

  expect(merged).toEqual(['vue', 'react', 'svelte', 'solid'])
})

// test.skip
// test.fails
test('deep merge with overlaps', () => {
  const merged = deepMerge(
    {
      name: 'Augusto',
      accounts: {
        github:'unkown'
      }
    },
    {
      accounts: {
        twitter: 'gtoborges_'
      }
    }
  )

  expect(merged).toMatchSnapshot()
})

test('throws errors on merging two different types', () => {
  expect(() => deepMerge(
    ['foo', 'bar'],
    { foo: 'bar' }
  )).toThrowError('Can not merge two different types')
})
