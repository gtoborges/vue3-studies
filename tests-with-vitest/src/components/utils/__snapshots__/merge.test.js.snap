// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`deep merge with overlaps 1`] = `
{
  "accounts": {
    "github": "unkown",
    "twitter": "gtoborges_",
  },
  "name": "Augusto",
}
`;
